### PR TITLE
ESQL: Grammar - FROM METADATA no longer require []

### PR DESCRIPTION
**Pull request by costin**
Remove usage of [ ] through-out the grammar, in this case inside
 FROM METADATA.

### CODE REVIEW

**Review by bpintea — APPROVED** [review]
Lgtm

**Review by luigidellaquila — APPROVED** [review]
Thanks Costin, LGTM.

For the failing tests (all the CSV tests that still use square brackets for metadata and that now expect a warning), my feeling is that there is no solution apart from skipping them for `v < 8.13`: warnings are checked as mandatory, so the tests will always fail in old versions, where the warning is not emitted

File: docs/changelog/105221.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+pr: 105221
+summary: "ESQL: Grammar - FROM METADATA no longer require []"
```
Comment (astefan) [inline docs/changelog/105221.yaml:2]: `require` -> `requires`

File: docs/changelog/105221.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+area: ES|QL
+type: breaking
+issues: []
+breaking:
+  title: "ESQL: Grammar - FROM METADATA no longer require []"
```
Comment (astefan) [inline docs/changelog/105221.yaml:7]: `require` -> `requires`

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java [hunk 3]
```diff
@@ -180,7 +182,21 @@ public LogicalPlan visitFromCommand(EsqlBaseParser.FromCommandContext ctx) {
         TableIdentifier table = new TableIdentifier(source, null, visitFromIdentifiers(ctx.fromIdentifier()));
         Map<String, Attribute> metadataMap = new LinkedHashMap<>();
         if (ctx.metadata() != null) {
-            for (var c : ctx.metadata().fromIdentifier()) {
+            var deprecatedContext = ctx.metadata().deprecated_metadata();
+            MetadataOptionContext metadataOptionContext = null;
+            if (deprecatedContext != null) {
+                var s = source(deprecatedContext).source();
+                addWarning(
+                    "Line {}:{}: Square brackets '[]' need to be removed in FROM METADATA declaration",
```
Comment (astefan) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java:190]:
```suggestion
                    "Line {}:{}: Square brackets '[]' need to be removed from FROM METADATA declaration",
```

**Review by astefan — APPROVED** [review]
LGTM. Left some minor comments about wording.

**Comment by costin** [comment]
Thanks for the reviews folks - I've updated the test and raised #105267

**Review by fang-xing-esql — APPROVED** [review]
LGTM.

**Pull request merged**